src/main/java/games/strategy/engine/data/UnitCollection.java

```diff
@@ -7,14 +7,15 @@
 import java.util.Iterator;
 import java.util.List;
 import java.util.Set;
+import java.util.stream.Collectors;
 
 import games.strategy.util.IntegerMap;
 import games.strategy.util.Match;
 
 /**
  * A collection of units.
  */
-public class UnitCollection extends GameDataComponent implements Iterable<Unit> {
+public class UnitCollection extends GameDataComponent implements Collection<Unit> {
```

Comment:
I wasn't sure if this was going to cause a compatibility problem with old save games, but apparently interfaces are not serialized with the class descriptor (good to know).  I tried loading a 3635 save game, and it appeared to work fine.